# I. Data Loading

Memuat dataset transaksi e-commerce sebagai data mentah sebelum dilakukan data cleaning dan proses transformasi data

In [2]:
import pandas as pd

df = pd.read_csv("../data/global_ecommerce_sales.csv")

df.head()

,Order_ID,Order_Date,Customer_Name,Customer_Segment,Country,Region,Product_Category,Product_Name,Quantity,Unit_Price,Discount_Percent,Total_Sales,Shipping_Cost,Profit,Payment_Method
0,ORD-11121,2023-01-02,Karen Suzuki,Corporate,United States,North America,Technology,Wireless Bluetooth Headphones,3,99.43,0,298.29,9.31,124.92,Cash on Delivery
1,ORD-11244,2023-01-02,John Johansson,Corporate,Spain,Europe,Technology,Mechanical Gaming Keyboard,4,97.93,20,313.38,14.31,83.62,Cash on Delivery
2,ORD-10325,2023-01-03,Jessica Garcia,Consumer,Mexico,North America,Office Supplies,Binder Clips Assorted 48pc,2,10.74,0,21.48,8.12,3.69,Credit Card
3,ORD-10467,2023-01-03,Clara Taylor,Corporate,Italy,Europe,Technology,Webcam HD 1080p,2,61.86,15,105.16,10.79,26.32,PayPal
4,ORD-11454,2023-01-05,Felix Thomas,Consumer,Italy,Europe,Furniture,Standing Desk Converter,4,330.67,20,1058.14,11.09,253.44,Credit Card


Dataset berhasil dimuat sebagai data mentah yang akan diproses lebih lanjut melalui tahapan transformasi untuk memastikan kesiapan data sebelum dilakukan exploratory data analysis dan visualisasi dashboard.

# II. Convert order_date to Datetime

Mengubah tipe data kolom Order_Date dari object menjadi datetime agar dapat digunakan dalam analisis
berbasis waktu seperti tren penjualan bulanan, tahunan, maupun identifikasi pola musiman transaksi.

In [3]:
df["Order_Date"] = pd.to_datetime(df["Order_Date"])

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 15 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Order_ID          2000 non-null   object        
 1   Order_Date        2000 non-null   datetime64[ns]
 2   Customer_Name     2000 non-null   object        
 3   Customer_Segment  2000 non-null   object        
 4   Country           2000 non-null   object        
 5   Region            2000 non-null   object        
 6   Product_Category  2000 non-null   object        
 7   Product_Name      2000 non-null   object        
 8   Quantity          2000 non-null   int64         
 9   Unit_Price        2000 non-null   float64       
 10  Discount_Percent  2000 non-null   int64         
 11  Total_Sales       2000 non-null   float64       
 12  Shipping_Cost     2000 non-null   float64       
 13  Profit            2000 non-null   float64       
 14  Payment_Method    2000 n

Kolom Order_Date berhasil dikonversi dari tipe data object menjadi datetime, sehingga memungkinkan analisis tren
penjualan berbasis waktu seperti perbandingan performa penjualan antar bulan ataupun tahun.

Transformasi ini menjadi penting sebelum melakukan exploratory data analysis pada pola transaksi temporal.

# III. Create Time Based Features

Membuat variabel turunan berbasis waktu untuk mendukung analisis tren penjualan secara temporal.

In [5]:
df["Year"] = df["Order_Date"].dt.year
df["Month"] = df["Order_Date"].dt.month
df["Month_Name"] = df["Order_Date"].dt.month_name()

In [6]:
df.head()

,Order_ID,Order_Date,Customer_Name,Customer_Segment,Country,Region,Product_Category,Product_Name,Quantity,Unit_Price,Discount_Percent,Total_Sales,Shipping_Cost,Profit,Payment_Method,Year,Month,Month_Name
0,ORD-11121,2023-01-02,Karen Suzuki,Corporate,United States,North America,Technology,Wireless Bluetooth Headphones,3,99.43,0,298.29,9.31,124.92,Cash on Delivery,2023,1,January
1,ORD-11244,2023-01-02,John Johansson,Corporate,Spain,Europe,Technology,Mechanical Gaming Keyboard,4,97.93,20,313.38,14.31,83.62,Cash on Delivery,2023,1,January
2,ORD-10325,2023-01-03,Jessica Garcia,Consumer,Mexico,North America,Office Supplies,Binder Clips Assorted 48pc,2,10.74,0,21.48,8.12,3.69,Credit Card,2023,1,January
3,ORD-10467,2023-01-03,Clara Taylor,Corporate,Italy,Europe,Technology,Webcam HD 1080p,2,61.86,15,105.16,10.79,26.32,PayPal,2023,1,January
4,ORD-11454,2023-01-05,Felix Thomas,Consumer,Italy,Europe,Furniture,Standing Desk Converter,4,330.67,20,1058.14,11.09,253.44,Credit Card,2023,1,January


Kolom Year, Month, dan Month_Name berhasil dibuat untuk mendukung analisis tren penjualan berdasarkan waktu serta mempermudah proses agregasi data pada dashboard interaktif.

# IV. Create Profit_Margin Feature

Menghitung rasio profit terhadap total penjualan sebagai indikator efisiensi profitabilitas transaksi.

In [7]:
df["Profit_Margin"] = df["Profit"] / df["Total_Sales"]

In [8]:
df.head()

,Order_ID,Order_Date,Customer_Name,Customer_Segment,Country,Region,Product_Category,Product_Name,Quantity,Unit_Price,Discount_Percent,Total_Sales,Shipping_Cost,Profit,Payment_Method,Year,Month,Month_Name,Profit_Margin
0,ORD-11121,2023-01-02,Karen Suzuki,Corporate,United States,North America,Technology,Wireless Bluetooth Headphones,3,99.43,0,298.29,9.31,124.92,Cash on Delivery,2023,1,January,0.418787
1,ORD-11244,2023-01-02,John Johansson,Corporate,Spain,Europe,Technology,Mechanical Gaming Keyboard,4,97.93,20,313.38,14.31,83.62,Cash on Delivery,2023,1,January,0.266833
2,ORD-10325,2023-01-03,Jessica Garcia,Consumer,Mexico,North America,Office Supplies,Binder Clips Assorted 48pc,2,10.74,0,21.48,8.12,3.69,Credit Card,2023,1,January,0.171788
3,ORD-10467,2023-01-03,Clara Taylor,Corporate,Italy,Europe,Technology,Webcam HD 1080p,2,61.86,15,105.16,10.79,26.32,PayPal,2023,1,January,0.250285
4,ORD-11454,2023-01-05,Felix Thomas,Consumer,Italy,Europe,Furniture,Standing Desk Converter,4,330.67,20,1058.14,11.09,253.44,Credit Card,2023,1,January,0.239515


Kolom Profit_Margin menunjukkan rasio keuntungan terhadap total penjualan pada setiap transaksi. Nilai ini digunakan untuk mengukur efisiensi profitabilitas penjualan, sehingga memungkinkan identifikasi kategori produk, region, atau segmen pelanggan yang menghasilkan margin keuntungan lebih tinggi dibandingkan yang lain.

# V. Identify Negative Profit Transactions

Mengidentifikasi transaksi dengan nilai profit negatif untuk memahami potensi faktor yang menyebabkan kerugian pada penjualan.

In [9]:
# Melihat Transaksi Yang Rugi

df[df["Profit"] < 0]

,Order_ID,Order_Date,Customer_Name,Customer_Segment,Country,Region,Product_Category,Product_Name,Quantity,Unit_Price,Discount_Percent,Total_Sales,Shipping_Cost,Profit,Payment_Method,Year,Month,Month_Name,Profit_Margin
11,ORD-11499,2023-01-08,Kenji Rossi,Corporate,Spain,Europe,Office Supplies,Paper Clips Box 500pc,3,7.80,10,21.06,12.61,-2.08,Cash on Delivery,2023,1,January,-0.098765
14,ORD-10793,2023-01-13,Zara Brown,Corporate,Italy,Europe,Office Supplies,Tape Dispenser Heavy Duty,1,15.82,0,15.82,9.16,-0.46,PayPal,2023,1,January,-0.029077
20,ORD-10686,2023-01-15,Clara Smith,Corporate,Colombia,South America,Office Supplies,Whiteboard Markers Set 8,5,9.91,25,37.16,20.73,-5.87,Cash on Delivery,2023,1,January,-0.157966
23,ORD-10846,2023-01-17,Yuki Costa,Corporate,UAE,Middle East & Africa,Office Supplies,Sticky Notes Multicolor 6-Pack,2,11.41,5,21.68,15.26,-3.85,Bank Transfer,2023,1,January,-0.177583
31,ORD-11723,2023-01-23,Raj Costa,Consumer,United Kingdom,Europe,Office Supplies,Stapler Full-Size,1,8.54,15,7.26,7.97,-4.55,Credit Card,2023,1,January,-0.626722
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1954,ORD-10352,2025-12-08,Chen Patel,Corporate,Nigeria,Middle East & Africa,Office Supplies,Tape Dispenser Heavy Duty,2,13.26,10,23.87,15.22,-3.28,PayPal,2025,12,December,-0.137411
1967,ORD-11898,2025-12-15,Xiao Johansson,Home Office,Mexico,North America,Office Supplies,Sticky Notes Multicolor 6-Pack,1,9.32,0,9.32,6.23,-1.10,Cash on Delivery,2025,12,December,-0.118026
1968,ORD-10723,2025-12-16,Hanna Martinez,Corporate,Germany,Europe,Office Supplies,Paper Clips Box 500pc,4,6.29,10,22.64,14.16,-2.84,Credit Card,2025,12,December,-0.125442
1982,ORD-11286,2025-12-23,Robert Sato,Consumer,Canada,North America,Office Supplies,Ballpoint Pen Pack 12,2,9.45,15,16.06,9.31,-1.76,PayPal,2025,12,December,-0.109589


In [10]:
# Menhitung Jumlah Transaksi Rugi

(df["Profit"] < 0).sum()

272

In [11]:
# Menghitung Proporsinya

(df["Profit"] <0).mean()

0.136

In [12]:
df[df["Profit"] < 0][["Product_Category", "Discount_Percent", "Shipping_Cost"]]

,Product_Category,Discount_Percent,Shipping_Cost
11,Office Supplies,10,12.61
14,Office Supplies,0,9.16
20,Office Supplies,25,20.73
23,Office Supplies,5,15.26
31,Office Supplies,15,7.97
...,...,...,...
1954,Office Supplies,10,15.22
1967,Office Supplies,0,6.23
1968,Office Supplies,10,14.16
1982,Office Supplies,15,9.31


Ditemukan sebanyak 272 transaksi atau sekitar 13.6% dari total transaksi menghasilkan profit negatif. Sebagian besar transaksi yang merugi berasal dari kategori Office Supplies, yang menunjukkan kemungkinan rendahnya margin produk pada kategori tersebut. Selain itu, nilai Shipping_Cost yang relatif tinggi dibandingkan Total_Sales serta penerapan diskon pada beberapa transaksi diduga menjadi faktor utama yang berkontribusi terhadap kerugian penjualan. Temuan ini menunjukkan adanya potensi ketidakefisienan pada strategi pricing atau distribusi untuk kategori produk tertentu yang perlu dianalisis lebih lanjut pada tahap exploratory data analysis.

# VI. Final Dataset Verification

Memastikan dataset telah melalui proses transformasi yang diperlukan dan siap digunakan pada tahap
exploratory data analysis serta visualisasi dashboard.

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 19 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   Order_ID          2000 non-null   object        
 1   Order_Date        2000 non-null   datetime64[ns]
 2   Customer_Name     2000 non-null   object        
 3   Customer_Segment  2000 non-null   object        
 4   Country           2000 non-null   object        
 5   Region            2000 non-null   object        
 6   Product_Category  2000 non-null   object        
 7   Product_Name      2000 non-null   object        
 8   Quantity          2000 non-null   int64         
 9   Unit_Price        2000 non-null   float64       
 10  Discount_Percent  2000 non-null   int64         
 11  Total_Sales       2000 non-null   float64       
 12  Shipping_Cost     2000 non-null   float64       
 13  Profit            2000 non-null   float64       
 14  Payment_Method    2000 n

Dataset telah melalui proses transformasi tipe data pada variabel Order_Date serta penambahan variabel turunan seperti Year, Month, Month_Name, dan Profit_Margin. Variabel-variabel ini akan digunakan untuk mendukung analisis tren penjualan berbasis waktu serta evaluasi efisiensi profitabilitas transaksi pada tahap exploratory data analysis dan pembuatan dashboard.

# VII. Save Cleaned Dataset

Menyimpan dataset hasil transformasi sebagai sumber data utama Exploratory Data Analysis dan Visualisasi
dashboard tanpa perlu mengulang proses data cleaning

In [14]:
df.to_csv("../data/global_ecommerce_clean.csv", index=False)

Dataset hasil proses data cleaning berhasil disimpan sebagai file baru sehingga dapat digunakan sebagai sumber data utama pada tahap exploratory data analysis dan visualisasi dashboard secara konsisten dan efisien.